src/nvim/ex_cmds.c

```diff
@@ -5848,53 +5828,21 @@ void ex_helpclose(exarg_T *eap)
   }
 }
 
-/// Tries to enter to an existing window of given buffer. If no existing buffer
-/// is found, creates a new split.
-///
-/// @return  OK/FAIL.
-int sub_preview_win(buf_T *preview_buf)
-{
-  if (preview_buf != NULL) {
-    FOR_ALL_WINDOWS_IN_TAB(wp, curtab) {
-      if (wp->w_buffer == preview_buf) {
-        win_enter(wp, false);
-
-        return OK;
-      }
-    }
-  }
-  return win_split((int)p_cwh, WSP_BOT);
-}
-
 /// Shows the effects of the :substitute command being typed ('inccommand').
 /// If inccommand=split, shows a preview window and later restores the layout.
-static buf_T *show_sub(exarg_T *eap, pos_T old_cusr, PreviewLines *preview_lines, int hl_id,
-                       int src_id, handle_T bufnr)
+static void show_sub(exarg_T *eap, pos_T old_cusr, PreviewLines *preview_lines, int hl_id)
   FUNC_ATTR_NONNULL_ALL
 {
-  win_T *save_curwin = curwin;
-  cmdmod_T save_cmdmod = cmdmod;
   char_u *save_shm_p = vim_strsave(p_shm);
   PreviewLines lines = *preview_lines;
   buf_T *orig_buf = curbuf;
+  buf_T *preview_buf = NULL;
 
   // We keep a special-purpose buffer around, but don't assume it exists.
-  buf_T *preview_buf = bufnr ? buflist_findnr(bufnr) : 0;
-  cmdmod.split = 0;               // disable :leftabove/botright modifiers
-  cmdmod.tab = 0;                 // disable :tab modifier
-  cmdmod.noswapfile = true;       // disable swap for preview buffer
   // disable file info message
   set_string_option_direct("shm", -1, (char_u *)"F", OPT_FREE,
                            SID_NONE);
 
-  bool outside_curline = (eap->line1 != old_cusr.lnum
-                          || eap->line2 != old_cusr.lnum);
-  bool preview = outside_curline && (*p_icm != 'n');
-  if (preview_buf == curbuf) {  // Preview buffer cannot preview itself!
```

Comment:
is this guard still needed somewhere?

Reply:
Yes, the guard has been moved to `open_preview_win`